rllib/env/policy_server_input.py

```diff
@@ -176,16 +193,22 @@ def get_metrics(self) -> List[RolloutMetrics]:
 
     @override(InputReader)
     def next(self):
-        return self.samples_queue.get()
+        # Blocking wait until there is something in the deque.
+        while len(self.samples_queue) == 0:
+            time.sleep(0.1)
```

Comment:
its important to know that time.sleep is equivalent to yield in terms of letting go of control of the processor to other threads...